addons/web/models/ir_http.py

```diff
@@ -137,21 +137,38 @@ def _get_content_common(self, xmlid=None, model='ir.attachment', res_id=None, fi
             xmlid=xmlid, model=model, id=res_id, field=field, unique=unique, filename=filename,
             filename_field=filename_field, download=download, mimetype=mimetype, access_token=access_token
         )
+        placeholder_content = None
         if status != 200:
-            return self._response_by_status(status, headers, content)
-        else:
-            headers.append(('Content-Length', len(content)))
-            response = request.make_response(content, headers)
+            if filename and filename.endswith(".mp4") and not content:
+                # Show placeholder video if video should've 404'd
+                placeholder_content = self._video_placeholder()
+                headers.append(('Content-Type', "video/mp4"))
+            else:
+                return self._response_by_status(status, headers, content)
+
+        content = placeholder_content or (content if not isinstance(content, tuple) else "")
+        headers.append(('Content-Length', len(content)))
+
+        # Setting CSP required in order to play same origin videos as BG videos.
+        # It presents no additional security risk when set to 'self', for more info
+        # check https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Content-Security-Policy/media-src.
+        try:
+            previous_csp = headers.index(('Content-Security-Policy', "default-src 'none'"))
+            headers[previous_csp] = ('Content-Security-Policy', "media-src 'self'")
+        except ValueError:
+            headers.append(('Content-Security-Policy', "media-src 'self'"))
+        response = request.make_response(content, headers)
         return response
 
     @api.model
     def _content_image(self, xmlid=None, model='ir.attachment', res_id=None, field='raw',
             filename_field='name', unique=None, filename=None, mimetype=None, download=None,
             width=0, height=0, crop=False, quality=0, access_token=None, **kwargs):
+        allow_xaccel = not(width or height or crop or quality)
```

Comment:
This is written weirdly, `not` is not a function.

Reply:
Not is not a function, it's an operator reversing a boolean expression and "width or height or crop or quality" is a boolean expression, so this doesn't seem that weird to me. How would you have written it?

Reply:
> Not is not a function, it's an operator reversing a boolean expression and "width or height or crop or quality" is a boolean expression when they are all false, and still yields the expected result when either is true, so this doesn't seem that weird to me. How would you have written it?

With a space after the operator. That is normally the style for Python operators, or keywords in general.

> Also, not sure there is a fundamental difference between operators and functions

Operators can not be shadowed, and have bespoke evaluation rules (e.g. they can delay the evaluation of their parameters though that is not a factor here),

Reply:
So a more correct way to write it would be
`allow_xaccel = not bool(width or height or crop or quality)` or simply adding a space 
`allow_xaccel = not (width or height or crop or quality)`
?